cf-update-circl.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+#
+# cf-update-circl.py updates the CIRCL package in src/circl.
+import os
+import sys
+import tempfile
+import subprocess
+
+# Name of the CIRCL package as it will appear in the standard library.
+circl_name = 'circl'
+
+# The git repository form which CIRCL will be cloned.
+circl_repo = 'https://github.com/cloudflare/circl'
+
+# The git branch to checkout.
+circl_branch = 'master'
+
+# The set of CIRCL packages that the standard library depends on.
+circl_pkg_dir = [
+  'circl',
+  'circl/ecc/goldilocks',
+  'circl/dh/sidh',
+  'circl/dh/sidh/internal/common',
+  'circl/dh/sidh/internal/p434',
+  'circl/dh/sidh/internal/p503',
+  'circl/dh/sidh/internal/p751',
+  'circl/internal/conv',
+  'circl/internal/sha3',
+  'circl/internal/test',
+  'circl/math',
+  'circl/math/fp25519',
+  'circl/math/fp448',
+  'circl/math/mlsbset',
+  'circl/pki',
+  'circl/sign',
+  'circl/sign/dilithium/internal/common',
+  'circl/sign/dilithium/internal/common/params',
+  'circl/sign/dilithium/mode3',
+  'circl/sign/dilithium/mode3/internal',
+  'circl/sign/dilithium/mode4',
+  'circl/sign/dilithium/mode4/internal',
+  'circl/sign/ed25519',
+  'circl/sign/ed448',
+  'circl/sign/eddilithium3',
+  'circl/sign/eddilithium4',
+  'circl/sign/schemes',
+  'circl/simd/keccakf1600',
+]
+
+# The directories of test data required for CIRCL unit tests.
+circl_testdata_dir = [
+  'circl/dh/sidh/testdata',
+  'circl/internal/sha3/testdata',
+  'circl/sign/ed25519/testdata',
+  'circl/sign/ed448/testdata',
+]
+
+# Miscellaneous files to copy from CIRCL.
+circl_misc_file = [
+  'circl/LICENSE',
+]
+
+src = os.path.join(os.path.dirname(os.path.abspath(sys.argv[0])), 'src')
+src_circl = os.path.join(src, circl_name)
+
+# Returns true if fn is a source file for CIRCL. Currently this includes Go
+# sources code, C header files, and ASM source code.
+def is_pkg_file(fn):
+    ok = False
+    for suffix in ['go', 'h', 's']:
+      ok |= fn.endswith('.' + suffix)
+    return ok
+
+# Modifies file fn by replacing imports with the corresponding package in the
+# standard library.
+def handle_pkg_file(fn):
+    # Fix Go imports.
+    if fn.endswith('.go'):
+        with open(fn, 'rb') as f:
+            s = f.read()
+        s = s.replace(
+            b'github.com/cloudflare/circl',
+            bytes(circl_name, 'ascii'),
```

Comment:
All .go files are expected to be UTF-8, so you might as well use `open(fn, 'r')` and `open(fn, 'w')` and drop the `b` prefixes.

If you do not, usually in Python one uses `circl_name.encode('utf8')` instead of `bytes(circl_name, 'utf8')`.

Reply:
Dropped `b` prefixes and changed `'ascii'` to `'utf8'` below.